### DISCUSSION TITLE
Stable version

### DISCUSSION THREAD

**Discussion by AlexanderLanin** [ccache/ccache#773] [Development]
Proposal: at ccache 4.2 split into feature and stabilization branches as done before with e.g. 3.7
I guess this was simply missed with ccache 4.0

There should probably be a branch that grows more mature for real life use (due to delayed feedback from users) (4.2.x) and one for more new cool features (4.3)

**Comment by jrosdahl** [comment]
I have intentionally avoided creating maintenance branches for 4.0 and 4.1 since I feel that the so far applied features have low risk of introducing regressions. master is in my mind still the stabilization branch for the 4.0 release but with some minor features sprinkled on top. The reason for not doing release branches anyway is to avoid having to cherry-pick fixes to a release branch all the time. A process optimization, that is. As soon as a major feature lands on master I'll branch. (This is also how I used to do it for 3.x versions, so it's nothing new.)

Does this sound reasonable?

**Reply by AlexanderLanin** [reply]
Yes reasonable. With current activity on ccache it's easier to maintain fewer branches.

**Reply by ftrofin** [reply]
Joel, what is the process for creating release tags? Some important bugs were fixed in main since the 4.2 release tag was created and I was wondering when we will have a new release tag that incorporates all those bug fixes. Is it on a fixed/date schedule?

**Reply by jrosdahl** [reply]
@ftrofin: A new release is created when enough interesting/important features or bug fixes have accumulated and I have enough spare time and motivation to create a release. It's not based on a time schedule. Note that creating a tag is just a small part of the release work.

In this case, I haven't created a 4.2.1 release since I haven't considered any of the bug fixes important enough to hurry a release. Which bug fixes do you consider important?

**Reply by ftrofin** [reply]
Can't recall exactly but I remember seeing some conversations about cache misses in certain situations (quick search of closed PRs did not help me find it). That's what prompted me to ask. Maybe I'm mixing up things in my head.
In any event, Joel thanks for clarifying how the releases work. I'll wait for the next release to come out.
